File: donghyo/Programmers/printer.py
from collections import deque

def solution(priorities, location):
    queue = deque([[p, idx] for idx, p in enumerate(priorities)])
    answer = 0

    while len(queue):
        valid = queue.popleft()

        if queue and valid[0] < max(queue)[0]:
            queue.append(valid)
        else:
            answer += 1
            if valid[1] == location:
                break
            
    print(answer)
    return answer

solution([2, 1, 3, 2], 2)

File: donghyo/Programmers/Printer.java
import java.util.*;
 
public class Printer {
    public static void main(String[] args){
        int[] priorities = {1, 1, 9, 1, 1, 1};
        int location = 0;
        Solution call = new Solution();
        call.solution(priorities, location);

    }
}

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 1;
        PriorityQueue<Integer> maxHeap = new PriorityQueue<>(Collections.reverseOrder());
        
        for(Integer e : priorities){
            maxHeap.add(e);
        }

        while(!maxHeap.isEmpty()){
            for(int idx = 0; idx < priorities.length; idx++){
                int temp = maxHeap.peek();

                if(priorities[idx] == temp){
                    System.out.println(temp);
                    if(idx == location){
                        System.out.println(answer);
                        return answer;
                    }
                    maxHeap.poll();
                    answer++;
                }
                
            }
        }
        return answer;
    }
}
